vim-engine/src/main/kotlin/com/maddyhome/idea/vim/handler/MotionActionHandler.kt

```diff
@@ -83,6 +83,14 @@ public sealed class MotionActionHandler : EditorActionHandlerBase(false) {
     ): Motion
   }
 
+  /**
+   * Support for commands that can be executed either once or for each caret depending on some circumstances
+   * TODO this class should not exist at all, changes to command execution are required
+   */
+  public abstract class AmbiguousExecution : MotionActionHandler() {
```

Comment:
Well, I don't like the name, but I don' have a better suggestion.

If `AmbiguousExecution` is not supposed to override functions from `MotionActionHandler`, don't you want to `final override` them?

Also, I faced such issue before and made a bit different approach to this. See `ConditionalSingleExecution`.

Reply:
I don't like both `AmbiguousExecution` and `ConditionalSingleExecution` TBH. We should gather up and figure out how to avoid using such dirty hacks